src/insights_client/tests/test_motd.py

```diff
@@ -0,0 +1,116 @@
+            patch = unittest.mock.patch(target, mocked_path)
+            self.patches.append(patch)
+            patch.start()
+
+        with (self.chroot / "etc/insights-client/insights-client.motd").open("w") as f:
+            f.write("This is the content of the MOTD file.")
+
+    @property
+    def chroot(self) -> pathlib.Path:
+        return pathlib.Path(self._dir.name)
+
+    def __del__(self):
+        for patch in self.patches:
+            patch.stop()
+        self._dir.cleanup()
+
+
+@pytest.fixture(scope="function")
+def mock_fs():
+    fs = MockFS()
+    yield fs
+    del fs
+
+
+def test_present(mock_fs):
+    assert not (mock_fs.chroot / "etc/motd.d/insights-client").exists()
+
+    # The file gets created/symlinked when .registered & .unregistered do not exist
+    insights_client.update_motd_message()
+    assert (mock_fs.chroot / "etc/insights-client/insights-client.motd").samefile(
+        mock_fs.chroot / "etc/motd.d/insights-client"
+    )
+    assert (mock_fs.chroot / "etc/motd.d/insights-client").exists()
+
+
+def test_absent_on_dot_registered(mock_fs):
+    # The MOTD file exists by default...
+    insights_client.update_motd_message()
+    assert (mock_fs.chroot / "etc/motd.d/insights-client").exists()
+    # ...and gets removed when .registered exists.
+    (mock_fs.chroot / "etc/insights-client/.registered").open("w").close()
+
+    insights_client.update_motd_message()
+    assert not (mock_fs.chroot / "etc/motd.d/insights-client").exists()
+
+    # It stays absent when run multiple times.
+    insights_client.update_motd_message()
+    assert not (mock_fs.chroot / "etc/motd.d/insights-client").exists()
+
+
+def test_absent_on_dot_unregistered(mock_fs):
+    # The MOTD file exists by default...
+    insights_client.update_motd_message()
+    assert (mock_fs.chroot / "etc/motd.d/insights-client").exists()
+    # ...and gets removed when .unregistered exists.
+    (mock_fs.chroot / "etc/insights-client/.unregistered").open("w").close()
+
+    insights_client.update_motd_message()
+    assert not (mock_fs.chroot / "etc/motd.d/insights-client").exists()
+
+    # It stays absent when run multiple times.
+    insights_client.update_motd_message()
+    assert not (mock_fs.chroot / "etc/motd.d/insights-client").exists()
+
+
+def test_ignored_on_dev_null(mock_fs):
+    # When the /etc/motd.d/insights-client is a symbolic link to /dev/null...
+    (mock_fs.chroot / "etc/motd.d/insights-client").symlink_to(pathlib.Path(os.devnull))
+
+    # ...it should not be overwritten...
+    insights_client.update_motd_message()
+    assert (mock_fs.chroot / "etc/motd.d/insights-client").samefile(
+        pathlib.Path(os.devnull)
+    )
+
+    # ...whether the MOTD file should be present or not.
+    (mock_fs.chroot / "etc/insights-client/.registered").open("w").close()
+    insights_client.update_motd_message()
+    assert (mock_fs.chroot / "etc/motd.d/insights-client").samefile(
```

Comment:
A variable may come handy, saving not only repetition, but also simplifying syntax.

```suggestion
    motd_file = mock_fs.chroot / "etc/motd.d/insights-client"
    motd_file.symlink_to(pathlib.Path(os.devnull))

    # ...it should not be overwritten...
    insights_client.update_motd_message()
    assert motd_file.samefile(pathlib.Path(os.devnull)

    # ...whether the MOTD file should be present or not.
    (mock_fs.chroot / "etc/insights-client/.registered").open("w").close()
    insights_client.update_motd_message()
    assert motd_file.samefile(
```

Reply:
I like to be explicit.